include/zephyr/drivers/flash/st_flash_api_extensions.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_FLASH_STM32_WRITE_PROTECT)
+/*
+ * STM32 sector write protection control.
+ *
+ * As an input this operation takes a structure with two sector masks,
+ * first mask is used to enable protection on sectors, while second mask
+ * is used to do the opposite. If both masks are 0, then protection will
+ * remain unchanged. If same sector is set on both masks, protection will
+ * be enabled.
+ *
+ * As an output, sector mask with enabled protection is returned.
+ * Input can be NULL if we only want to get protected sectors.
+ * Output can be NULL if not needed.
+ */
+#define FLASH_STM32_EX_OP_SECTOR_WP 0x8000
+struct flash_stm32_ex_op_sector_wp_in {
+	uint32_t enable_mask;
+	uint32_t disable_mask;
+};
+
+struct flash_stm32_ex_op_sector_wp_out {
+	uint32_t protected_mask;
+};
+#endif /* CONFIG_FLASH_STM32_WRITE_PROTECT */
+
+#if defined(CONFIG_FLASH_STM32_READOUT_PROTECTION)
+/*
+ * STM32 sector readout protection control.
+ *
+ * As an input this operation takes structure with information about desired
+ * RDP state. As an output the status after applying changes is returned.
+ */
+#define FLASH_STM32_EX_OP_RDP 0x8001
```

Comment:
Do we want to have EX_OP definitions Kconfig dependent?
Shouldn't we also have all defines grouped together to allow easier to enum existing and add new?

Reply:
Moved definitions to the enum.